app/views/spree/admin/orders/_invoice_table2.html.haml

```diff
@@ -17,6 +17,9 @@
       %tr
         %td
           = render 'spree/shared/line_item_name', line_item: item
+          %br
+          %small
+            %em= raw(item.variant.product.supplier.name)
```

Comment:
any chance we could pass something like `supplier_name` to this view? It'd reduce coupling a great deal but I'm not familiar with this view at all though.

Reply:
This is actually within a loop. The only variable for the template is `@order` and this template then goes through the items:
```rb
@order.line_items...each do |item|
  #...
end
```

We could put the content of the loop in a separate template, but I don't see the benefit here.

Reply:
I see, a very important point. I always hesitate to modify Spree code though. I'm also wondering if it's the responsibility of a line item to know the supplier name. Two ideas:

- Provide `LineItem#supplier` via the decorator. Less knowledge for the line item and the view can select what to display from the supplier: `raw(item.supplier.name)`.
- Or provide a helper that can select the supplier name of a line item, variant or product: `supplier_name(item)`.

If LineItem was our class I would definitely go for the first option, but since it's Spree, I'm not sure. What do you think?